sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/logging/WarningLogger.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.opentelemetry.exporter.implementation.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.util.annotation.Nullable;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class WarningLogger {
+
+    private final Logger logger;
+    private final String message;
+    private final AtomicLong numWarnings = new AtomicLong();
```

Comment:
I don't know how often this type will be instantiate but if it is often using a volatile and [`Atomic](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLongFieldUpdater.html), same for the AtomicBoolean below but with https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.html, will reduce the memory overhead of this object

Reply:
there are only a few of these instantiated at startup (similar to a Logger) so I think ok here